bittensor/btlogging/loggingmachine.py

```diff
@@ -107,14 +107,31 @@ def __init__(self, config: bittensor.config, name: str = BITTENSOR_LOGGER_NAME):
         #
         # In the future, we may want to add options to introduce other handlers
         # for things like log aggregation by external services.
-        self._handlers = self._configure_handlers(config)
+        self._handlers = self._configure_handlers(self._config)
 
         # configure and start the queue listener
         self._listener = self._create_and_start_listener(self._handlers)
 
         # set up all the loggers
         self._logger = self._initialize_bt_logger(name)
         self.disable_third_party_loggers()
+        self._enable_initial_state(self._config)
+
+    def _enable_initial_state(self, config):
+        """Set correct state action on initializing"""
+        if config.trace:
+            self.enable_trace()
+        elif config.debug:
```

Comment:
This `elif` would only enable debug if trace is not enabled. Is that what we want?

Reply:
Yes, that is the intended behavior. At a time, we can enable 1 exclusive logging state—Trace, Debug, or Default—based on the configuration provided.

**Trace**: This is the most verbose level, logging all messages including trace level messages. This has a level of 5. It contains all logging levels i.e. trace + debug + default

**Debug**: This has a level of 10. It contains debug + default logging levels. 

**Default**: This has a level of 20. It only contains default levels i.e. info, error, warning, success